### ISSUE TITLE
Reproduce issue

### ISSUE THREAD

**Issue by JiawLin** [RSer-XDU/G-FSDet#7]
Hello, I am trying to reproduce your experiment using your code and data, but I found that my results are different from yours. Could you please help me understand why?
your results:
![4db71f24b4abe04e148766393d5e660](https://github.com/RSer-XDU/G-FSDet/assets/137050210/f53223bc-794f-4c23-a2ec-841271749834)
The results obtained using my own trained base training checkpoint:
![30f8ea86e08292bcaccabde2e7a8289](https://github.com/RSer-XDU/G-FSDet/assets/137050210/7221c0e9-6b3e-42ac-9845-14f77ee99b98)
The results obtained using the base training checkpoint you provided:
![38bc762a7c6ea0726eb23d955bf28e1](https://github.com/RSer-XDU/G-FSDet/assets/137050210/49725a35-a25f-4dfb-9210-bf36be1eca84)
[my_results.zip](https://github.com/RSer-XDU/G-FSDet/files/11786470/my_results.zip)

**Comment by imyjx** [comment]
I have met with the same problem and my reproduce results is different from both of the author's and yours.
Are the other results, such as 3-,5-,and 20-shot can match the results on the paper?

**Comment by imyjx** [comment]
Sry, I cannot see your results because the fig is somehow blank. Here are my reproduction results on DIOR S1.
![image](https://github.com/RSer-XDU/G-FSDet/assets/26484929/b07872c9-b5c8-4f3e-ac83-05049a37ee0c)

**Comment by JiawLin** [comment]
I'm sorry, I don't know why my picture turned black, so I’ll send you an attachment, this is my reproduction result,  and I saw that my results on DIOR SI are better than yours.
















At 2023-07-05 15:19:33, "Yu JX" ***@***.***> wrote:

Sry, I cannot see your results because the fig is somehow blank. Here are my reproduction results on DIOR S1.

**Comment by imyjx** [comment]
> I'm sorry, I don't know why my picture turned black, so I’ll send you an attachment, this is my reproduction result, and I saw that my results on DIOR SI are better than yours. At 2023-07-05 15:19:33, "Yu JX" ***@***.***> wrote: Sry, I cannot see your results because the fig is somehow blank. Here are my reproduction results on DIOR S1. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you modified the open/close state.Message ID: ***@***.***>

Sry, I haven't receive your attachment. BTW, have you changed any hyperparameter, e.g. iters, lr issues, etc.?

**Comment by JiawLin** [comment]
> 
Sorry, I am a novice, this is my first time to ask questions on github, so I don’t know that using email reply will cause these problems, I did not change any hyperparameter in the reproduction process, but I used two gpu, the following are my reproduction results.
![image](https://github.com/RSer-XDU/G-FSDet/assets/137050210/9e23d903-c606-46be-8ac7-18daba3b065e)

[results.xlsx](https://github.com/RSer-XDU/G-FSDet/files/12257754/results.xlsx)

**Comment by JiawLin** [comment]
> > I'm sorry, I don't know why my picture turned black, so I’ll send you an attachment, this is my reproduction result, and I saw that my results on DIOR SI are better than yours. At 2023-07-05 15:19:33, "Yu JX" _**@**_._**> wrote: Sry, I cannot see your results because the fig is somehow blank. Here are my reproduction results on DIOR S1. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you modified the open/close state.Message ID: _**@**_.**_>
> 
> Sry, I haven't receive your attachment. BTW, have you changed any hyperparameter, e.g. iters, lr issues, etc.?

Sorry, I am a novice, this is my first time to ask questions on github, so I don’t know that using email reply will cause these problems, I did not change any hyperparameter in the reproduction process, but I used two gpu, the following are my reproduction results.
![image](https://github.com/RSer-XDU/G-FSDet/assets/137050210/e4f97ed4-a1d8-4a3d-a7a9-39876bd44c7c)

**Comment by RSer-XDU** [comment]
Thanks for your attention, which configuration are you running? 
It looks like your reported results are the performance of the ETF, not our final model.

![ETF](https://github.com/RSer-XDU/G-FSDet/assets/87374716/ac42aada-2d59-4617-b1fc-04fbd2b52531)
![Final](https://github.com/RSer-XDU/G-FSDet/assets/87374716/f1f91d6e-adc4-4a05-892b-3a89c547b703)

**Comment by YongZh958** [comment]
在我复现你的工作时，我发现与论文中的结果存在差异，希望你能够帮助我解决问题。
我使用的是单卡复现你的工作，所以将学习率设置为0.0005（默认值是0.001），其它参数并未修改。在基类上训练时，我训练的结果如下：
2023-11-13 18:34:36,484 - mmfewshot - INFO - Saving checkpoint at 36000 iterations
2023-11-13 18:34:38,620 - mmfewshot - INFO - Exp name: tfa_r101_fpn_dior_split1_base-training.py
2023-11-13 18:34:38,620 - mmfewshot - INFO - Iter [36000/36000] lr: 5.000e-05, eta: 0:00:00, time: 0.267, data_time: 0.013, memory: 3854, loss_rpn_cls: 0.0019, loss_rpn_bbox: 0.0120, loss_cls: 0.0414, acc: 98.3496, loss_bbox: 0.0897, loss: 0.1449
2023-11-13 18:48:33,665 - mmfewshot - INFO - 
+-------------------------+-------+-------+--------+-------+
| class                   | gts   | dets  | recall | ap    |
+-------------------------+-------+-------+--------+-------+
| airplane                | 8212  | 5196  | 0.572  | 0.542 |
| airport                 | 666   | 2087  | 0.773  | 0.629 |
| Expressway-toll-station | 688   | 702   | 0.513  | 0.508 |
| harbor                  | 3105  | 8866  | 0.751  | 0.608 |
| groundtrackfield        | 1885  | 2289  | 0.769  | 0.660 |
| dam                     | 538   | 2489  | 0.734  | 0.508 |
| golffield               | 575   | 1691  | 0.847  | 0.751 |
| storagetank             | 23361 | 16944 | 0.586  | 0.537 |
| tenniscourt             | 7343  | 6909  | 0.783  | 0.724 |
| vehicle                 | 26640 | 23684 | 0.454  | 0.425 |
| Expressway-Service-area | 1085  | 1782  | 0.576  | 0.540 |
| overpass                | 1782  | 4188  | 0.608  | 0.512 |
| stadium                 | 672   | 1057  | 0.589  | 0.442 |
| trainstation            | 509   | 2590  | 0.756  | 0.549 |
| windmill                | 2998  | 3553  | 0.785  | 0.720 |
+-------------------------+-------+-------+--------+-------+
| mAP                     |       |       |        | 0.577 |
+-------------------------+-------+-------+--------+-------+
2023-11-13 18:48:33,798 - mmfewshot - INFO - BASE_CLASSES_SPLIT1 mAP: 0.5770988464355469
2023-11-13 18:48:33,798 - mmfewshot - INFO - NOVEL_CLASSES_SPLIT1 mAP: nan
2023-11-13 18:48:33,861 - mmfewshot - INFO - Exp name: tfa_r101_fpn_dior_split1_base-training.py
2023-11-13 18:48:33,862 - mmfewshot - INFO - Iter(val) [11738]  AP50: 0.5770, BASE_CLASSES_SPLIT1: AP50: 0.5770, NOVEL_CLASSES_SPLIT1: AP50: nan, mAP: 0.5771

然后使用步骤二 初始化获得的权重
# step2: reshape the bbox head of base model for few shot fine-tuning
python -m tools.detection.misc.initialize_bbox_head \
    --src1 work_dirs/ETF_r101_fpn_voc-split1_base-training/latest.pth \
    --method randinit \
    --save-dir work_dirs/ETF_r101_fpn_voc-split1_base-training
最后我将power4_0.025_weight_0.5_alpha_tfa_r101_fpn_dior-split1_3shot-fine-tuning.py中
base_cpt = 'work_dirs/ETF_r101_fpn_voc-split1_base-training/base_model_random_init_bbox_head.pth',
load_from = (#'work_dirs/DIOR/TFA/tfa_r101_fpn_dior-split1_base-training/'
             'work_dirs/ETF_r101_fpn_voc-split1_base-training/'
             'base_model_random_init_bbox_head.pth')
修改为我的路径，同样将学习率改为0.0005，训练后获得结果如下：
2023-11-13 21:42:19,332 - mmfewshot - INFO - Saving checkpoint at 6000 iterations
2023-11-13 21:42:20,866 - mmfewshot - INFO - Exp name: power4_0.025_weight_0.5_alpha_tfa_r101_fpn_dior-split1_3shot-fine-tuning.py
2023-11-13 21:42:20,866 - mmfewshot - INFO - Iter [6000/6000]   lr: 5.000e-04, eta: 0:00:00, time: 0.205, data_time: 0.012, memory: 853, loss_rpn_cls: 0.0022, loss_rpn_bbox: 0.0020, loss_cls: 0.0298, acc: 99.8105, loss_bbox: 0.0480, loss_kd: 0.0305, loss_base_margin: 0.0052, loss_novel_margin: 0.0043, loss_neg_margin: 0.0063, loss: 0.1283, grad_norm: 1.1777
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 11738/11738, 13.9 task/s, elapsed: 843s, ETA:     0s


---------------iou_thr: 0.5---------------
2023-11-13 21:56:52,139 - mmfewshot - INFO - 
+-------------------------+-------+-------+--------+-------+
| class                   | gts   | dets  | recall | ap    |
+-------------------------+-------+-------+--------+-------+
| airplane                | 8212  | 6598  | 0.640  | 0.624 |
| airport                 | 666   | 6044  | 0.853  | 0.631 |
| Expressway-toll-station | 688   | 1825  | 0.568  | 0.514 |
| harbor                  | 3105  | 17000 | 0.777  | 0.600 |
| groundtrackfield        | 1885  | 3295  | 0.824  | 0.723 |
| dam                     | 538   | 4500  | 0.764  | 0.465 |
| golffield               | 575   | 2584  | 0.863  | 0.740 |
| storagetank             | 23361 | 16904 | 0.588  | 0.538 |
| tenniscourt             | 7343  | 8150  | 0.802  | 0.793 |
| vehicle                 | 26640 | 13554 | 0.385  | 0.355 |
| Expressway-Service-area | 1085  | 7825  | 0.640  | 0.553 |
| overpass                | 1782  | 9549  | 0.646  | 0.500 |
| stadium                 | 672   | 1066  | 0.652  | 0.495 |
| trainstation            | 509   | 9988  | 0.804  | 0.469 |
| windmill                | 2998  | 4623  | 0.804  | 0.772 |
| baseballfield           | 3434  | 15454 | 0.385  | 0.194 |
| basketballcourt         | 2146  | 14087 | 0.410  | 0.193 |
| bridge                  | 2589  | 17610 | 0.317  | 0.144 |
| chimney                 | 1031  | 7069  | 0.616  | 0.414 |
| ship                    | 35186 | 26707 | 0.076  | 0.091 |
+-------------------------+-------+-------+--------+-------+
| mAP                     |       |       |        | 0.490 |
+-------------------------+-------+-------+--------+-------+
2023-11-13 21:56:52,151 - mmfewshot - INFO - BASE_CLASSES_SPLIT1 mAP: 0.584722638130188
2023-11-13 21:56:52,151 - mmfewshot - INFO - NOVEL_CLASSES_SPLIT1 mAP: 0.2072904407978058
2023-11-13 21:56:52,229 - mmfewshot - INFO - Exp name: power4_0.025_weight_0.5_alpha_tfa_r101_fpn_dior-split1_3shot-fine-tuning.py
2023-11-13 21:56:52,230 - mmfewshot - INFO - Iter(val) [11738]  AP50: 0.4900, BASE_CLASSES_SPLIT1: AP50: 0.5850, NOVEL_CLASSES_SPLIT1: AP50: 0.2070, mAP: 0.4904
这与论文中的结果存在较大差异，请问我的步骤中出现了什么问题呢？非常感谢你的帮助并期待你的回信！

**Comment by clanguagesh** [comment]
hello, can you tell me where i can get the dior dataset
